packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -178,46 +163,39 @@ export class ConnectionStateHandler extends EventEmitterWithErrorHandling<IConne
 
         const oldState = this._connectionState;
         this._connectionState = value;
+        const quorum = this.handler.protocolHandler()?.quorum;
+        let client: ILocalSequencedClient | undefined;
+        if (this._clientId !== undefined) {
+            client = quorum?.getMember(this._clientId);
+        }
         if (value === ConnectionState.Connected) {
             assert(oldState === ConnectionState.Connecting,
                 0x1d8 /* "Should only transition from Connecting state" */);
             // Mark our old client should have left in the quorum if it's still there
-            if (this._clientId !== undefined) {
-                const client: ILocalSequencedClient | undefined =
-                    this.handler.protocolHandler()?.quorum.getMember(this._clientId);
-                if (client !== undefined) {
-                    client.shouldHaveLeft = true;
-                }
+            if (client !== undefined) {
```

Comment:
Same feedback here as in 0.38 change - this check is not sufficient, see shouldHaveLeft property.
Should we change API not to run into similar issues in the future? Feel free to open an issue for long term fix

Reply:
Sorry, I don't understand what you meant. I didn't changed the logic here. Just moved the get from quorum outside if.